osf/management/commands/backfill_domain_references.py

```diff
@@ -0,0 +1,96 @@
+# -*- coding: utf-8 -*-
+import re
+import logging
+import operator
+from functools import reduce
+
+from django.db.models import Q, OuterRef, Exists
+from django.apps import apps
+from django.core.management.base import BaseCommand
+from osf.external.spam.tasks import check_resource_for_domains
+from osf.models import DomainReference
+
+
+logger = logging.getLogger(__name__)
+
+DOMAIN_MATCH_REGEX = re.compile(r'(?P<protocol>\w+://)?(?P<www>www\.)?(?P<domain>[\w-]+\.\w+)(?P<path>/\w*)?')
+DOMAIN_SEARCH_REGEX = r'(http://[^ \'}\[\]\~\(\)\/]+|https://[^ \'}\[\]\~\(\)\/]+)'
```

Comment:
My one concern is still that these are matching very different things, so we're not going to get all of our back-references. Can we get some data about how different the results are (and how much longer the query takes) if we do something like `'r\w\s+\.\w+'`?